pytorch_lightning/profiler/pytorch.py

```diff
@@ -207,7 +170,7 @@ def __init__(
             self.on_train_start(local_rank=local_rank)
             self.on_train_start = super().on_train_start
```

Comment:
I don't understand the purpose of this. Why do we define `on_train_start` in this class if we are going to set it to the parents? Maybe the `on_train_start` function in this class should have a different name

Reply:
The user could want to use the profiler without Lightning and the functions need to be executed if the user provides a rank.

Reply:
I see. In any case, we could rename it to `initialize` so there's no need to hack the hooks